File: akk_example/src/main/java/com/hdu/chapter4_mailbox/DemoActor.java
package com.hdu.chapter4_mailbox;

import akka.actor.AbstractActor;
import akka.actor.PoisonPill;
import akka.actor.Props;
import akka.event.Logging;
import akka.event.LoggingAdapter;

/**
 * DESCRIPTION:
 *
 * @author shushoufu
 * @Date 2018/6/20
 * @Time 下午7:21
 */
public class DemoActor extends AbstractActor{
    LoggingAdapter log = Logging.getLogger(getContext().getSystem(),this);
    {
        for (Object msg:new Object[]{
                "lowpriority", "lowpriority",
                "highpriority", "pigdog", "pigdog2", "pigdog3", "highpriority",
                PoisonPill.getInstance()
        }){
            getSelf().tell(msg,getSelf());
        }
    }
    public static Props props(){
        return Props.create(DemoActor.class);
    }



    @Override
    public void postStop() throws Exception {
        super.postStop();
        log.info("demo actor stop...");
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .matchAny(msg->{
                    log.info(msg.toString());
                })
                .build();
    }
}

File: akk_example/src/main/scala/com/hdu/DemoActor.scala
package com.hdu

import akka.actor.{AbstractActor, Actor, Props}
import akka.japi.pf.ReceiveBuilder
import com.hdu.actors.SecondActor

/**
  * DESCRIPTION:
  *
  * @author shushoufu
  * @Date 2018/4/3
  * @Time 下午5:37
  */
object DemoActor{
  def props(msg:String):Props = Props(new DemoActor(msg))
}

class DemoActor(msg:String) extends Actor {
  var nextActor = context.actorOf(SecondActor.props(msg),"secondActor")
  override def receive: Receive = {
    case "hhh" =>
      println(String.format("receive msg is %s,hhh",msg))
    case "next" =>
      nextActor.tell("bob",self)
    case _ =>
      Thread.sleep(10000)
      System.out.println("===========>>>>>>>>>>>>>>>>")
  }
}
